src/pool.c

```diff
@@ -21,31 +21,27 @@ struct git_pool_page {
 
 static void *pool_alloc_page(git_pool *pool, size_t size);
 
-static size_t pool_system_page_size(void)
-{
-	static git_atomic_ssize cached_size = {0};
-	size_t page_size = 0;
+#ifndef GIT_DEBUG_POOL
 
-	if ((page_size = (size_t)git_atomic_ssize_get(&cached_size)) == 0) {
-		if (git__page_size(&page_size) < 0)
-			page_size = 4096;
-		/* allow space for malloc overhead */
-		page_size -= (2 * sizeof(void *)) + sizeof(git_pool_page);
-		git_atomic_ssize_set(&cached_size, (int64_t)page_size);
-	}
+static size_t system_page_size = 0;
 
-	return page_size;
+int git_pool_system_page_size_global_init(void)
```

Comment:
Let's make this the init function for the whole of the pool subsystem.  If we ever need to do more, it can be here.  This lets us be a little more general with the name of the init function, and a little briefer as well.

```suggestion
int git_pool_global_init(void)
```